pkg/controller/workload/workload_stream_test.go

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package workload
+
+import (
+	"context"
+	"errors"
+	"net"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/agiledragon/gomonkey/v2"
+	config_cluster_v3 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	discoveryv3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+type MockDiscovery struct {
+	Listener       *bufconn.Listener
+	responses      chan *discoveryv3.DiscoveryResponse
+	deltaResponses chan *discoveryv3.DeltaDiscoveryResponse
+	close          chan struct{}
+}
+
+func NewMockServer(t *testing.T) *MockDiscovery {
+	s := &MockDiscovery{
+		close:          make(chan struct{}),
+		responses:      make(chan *discoveryv3.DiscoveryResponse),
+		deltaResponses: make(chan *discoveryv3.DeltaDiscoveryResponse),
+	}
+
+	buffer := 1024 * 1024
+	listener := bufconn.Listen(buffer)
+	grpcServer := grpc.NewServer()
+	discoveryv3.RegisterAggregatedDiscoveryServiceServer(grpcServer, s)
+	go func() {
+		if err := grpcServer.Serve(listener); err != nil && !(err == grpc.ErrServerStopped || err.Error() == "closed") {
+			return
+		}
+	}()
+	t.Cleanup(func() {
+		grpcServer.Stop()
+		close(s.close)
+	})
+	s.Listener = listener
+	return s
+}
+
+func (f *MockDiscovery) StreamAggregatedResources(server discoveryv3.AggregatedDiscoveryService_StreamAggregatedResourcesServer) error {
+	numberOfSends := 0
```

Comment:
what is this value for? I do not see it's checked

Reply:
It is refer to https://github.com/istio/istio/blob/master/pilot/test/xdstest/mock_discovery.go